src/main/java/com/zontik/groshiky/service/TransactionService.java

```diff
@@ -34,15 +30,35 @@ public Transaction addTransaction(Transaction transaction, Account account) {
 
     @Override
     public void deleteTransactionById(Integer id) {
-        Transaction transaction= transactionRepository.getOne(id);
+        Transaction transaction = transactionRepository.findById(id)
+                .orElseThrow(() -> new MissionTransactionException("Unable to find transaction with id " + id));
         Account account = transaction.getAccount();
-        if(transaction.getTransactionType()==TransactionType.INCOME){
-            account.setBalance(account.getBalance()-transaction.getAmount());
-        }else{
-            account.setBalance(account.getBalance()-transaction.getAmount());
+        if (transaction.getTransactionType() == TransactionType.INCOME) {
```

Comment:
You've replaced == to equals in line 21. Probably it worths using the same approach everywhere.